compass/landice/tests/mismipplus/spin_up/__init__.py

```diff
@@ -0,0 +1,58 @@
+from compass.config import CompassConfigParser
+from compass.landice.tests.mismipplus.run_model import RunModel
+from compass.landice.tests.mismipplus.setup_mesh import SetupMesh
+from compass.testcase import TestCase
+
+
+class SpinUp(TestCase):
+    """
+    Test case for create the MISMIP+ mesh, initial conditions,
+    input files, and runs a short segment of the spin up experiments
+    """
+    def __init__(self, test_group):
+        """
+        Create the test case
+
+        Parameters
+        ----------
+        test_group : compass.landice.test.mismipplus
+            The test group that this test case belongs to
+
+        """
+        name = "spin_up"
+
+        super().__init__(test_group=test_group, name=name)
+
+        config = CompassConfigParser()
+        module = 'compass.landice.tests.mismipplus.spin_up'
+        # add from config
+        config.add_from_package(module, 'spin_up.cfg')
```

Comment:
I just wanted to chime in here with a little context as to why this is here. I don't think this section is necessary in most cases, but this was my work around to get the resolution (as passed in the config file) in the directory structure. Adding the config here allows the resolution to be parsed before the step is initialized, and therefore the resolution value can be passed to the `subdir` argument in the `Step` initialization. 

Maybe there's a better way to do this, but this was just what I came up with from looking through the compass repo (with a particular focus on the mesh convergence tests which has resolution in the directory structure).

Reply:
@andrewdnolan , thanks for the explanation - that's really helpful.  So what would happen if the user is specifying a cfg file in `compass setup` that has a different value from the default here?

Reply:
Then an error would be raised. At the time of `compass setup` the `resolution` value is stored as an attribute of the step. Then at compass run, the `resolution` is read from the config again and if the value from the config differs from the stored value an error is raised: 

 https://github.com/andrewdnolan/compass/blob/d0b93c4f0b5fe7a7e701aac0964989bb8d3a6343/compass/landice/tests/mismipplus/setup_mesh.py#L85-L97

The error should only pertain to `resolution` since it's the only one stored as an attribute. Other parameters (e.g. gutter, density) should have no problem if changed at runtime.

Reply:
I guess my question is if as written it is even possible to change things at setup time - if a user supplied a modified cfg file with `-f`, whatever value in there is ignored in preference for the version of the cfg file that is actually sitting in the repo, right?

Reply:
Hmm, I guess I'm not 100% clear by what you mean at "setup time."

For parameters in the config file, excluding `resolution`, it is possible to change values between running `compass setup` and `compass run`. I did some crude testing by printing to the log file parameter values changed between these two steps (e.g. `ice_density=200` from reference of `918`) and can confirm the values are updated accordingly. 

For resolution is it not. While that does make setting up various resolutions a little cumbersome, I think this limitation is needed since the directory structure is only created at the `compass setup` stage. If we didn't apply this restriction we could have a 2000m mesh sitting in the directory named 8000m, for example.

Reply:
Compass supports a 'user' cfg file with `-f` that can contain any cfg sections they want, e.g. specifying the allocation to submit jobs against plus some custom test case settings.  I tried doing that and got incorrect behavior like this:
```
cd compass/landice/tests/mismipplus/spin_up
cp spin_up.cfg blah.cfg
edit blah.cfg: resolution = 4000.
compass setup -n 95 -w $scratch5/compass-mismip+-spinup-blahcfg -f blah.cfg
ls $scratch5/compass-mismip+-spinup-blahcfg/landice/mismipplus/spin_up
```
and I see:
```
8000m  job_script.sh  load_compass_env.sh  spin_up.cfg  test_case.pickle
```
In other words, hardcoding retrieving the cfg from the repo eliminates the ability to use a user-specified cfg file with `-f`.  (I realize now I might not have made the question clear.)  I think you just want to change this bit of code to use self.config, which will already be populated with the merged cfg from the defaults in the repo plus the user one.

Reply:
Thanks for the clarification about the `-f` option to `compass setup`, I wasn't really familiar with it. 

I've revamp the way the directory structure is created, mainly by overloading the `compass.TestCase`'s `configure` method. Doing things this way ensures configuration passed over the command line (via `-f`) are properly resolved and accounted for. 

Would be great for you to experiment and corroborate that the fix works on your end. This overhaul was added in [4b0f35a](https://github.com/MPAS-Dev/compass/pull/737/commits/4b0f35a01e4606b80d1b86e37910645cfd780e92).